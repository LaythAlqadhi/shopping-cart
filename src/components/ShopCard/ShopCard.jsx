import { useState } from 'react';
import PropTypes from 'prop-types';
import css from './ShopCard.module.css';

const ShopCard = ({
  imgSrc,
  title,
  price,
  rating,
  onClick
}) => {
  return (
    <div className={css.container}>
      <img
        src={imgSrc}
        alt=""
        className={css.img}
      />
      <p>{title}</p>
      <div className={css.innerContainer}>
        <p>${price}</p>
        <p className={css.rating}>
          {rating}
          <span className="material-symbols-outlined">
            star
          </span>
        </p>
      </div>
      <button className={css.button} onClick={onClick}>
        Add to cart
      </button>
    </div>
  );
}

  ShopCard.propTypes = {
  imgSrc: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  rating: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired
}

export default ShopCard;