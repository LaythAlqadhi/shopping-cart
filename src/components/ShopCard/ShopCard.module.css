.container {
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
  width: 300px;
  height: 450px;
  padding: 25px;
  border: 2px solid var(--primary-accent-color);
  border-radius: 20px;
}

.innerContainer {
  width: 100%;
  padding: 12px;
  display: flex;
  justify-content: space-evenly;
  align-items: center;
}

.innerContainer p {
  font-weight: 600;
}

.img {
  width: 50%;
  margin-bottom: 25px;
}

.rating {
  display: flex;
  justify-content: center;
  align-items: center;
}

.rating span {
  font-size: 1em;
  font-weight: 600;
}