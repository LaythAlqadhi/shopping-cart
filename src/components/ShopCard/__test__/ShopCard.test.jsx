import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import ShopCard from '../ShopCard';

describe('ShopCard component', () => {
  it('renders in the document', () => {
    const { container } = render(<ShopCard />);

    expect(container).toMatchSnapshot();
  })
})