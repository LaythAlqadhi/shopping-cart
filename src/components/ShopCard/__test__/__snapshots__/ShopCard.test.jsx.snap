// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ShopCard component > renders in the document 1`] = `
<div>
  <div
    class="_container_87f6bf"
    data-status=""
  >
    <img
      alt=""
      class="_img_87f6bf"
    />
    <p />
    <div
      class="_innerContainer_87f6bf"
    >
      <p>
        $
      </p>
      <p
        class="_rating_87f6bf"
      >
        <span
          class="material-symbols-outlined"
        >
          star
        </span>
      </p>
    </div>
    <button
      class="_button_87f6bf"
    >
      Add to cart
    </button>
  </div>
</div>
`;
